### DISCUSSION TITLE
Behavior for components with fitContent.

### DISCUSSION THREAD

**Discussion by aidamag** [dxc-technology/halstack-react#1304] [Ideas] [answered]
**Clarification**
Talking on the base that `fitContent` has value `fit-content` (without `unset` as we have currently).

**Discussion**
I want to discuss the behaviour of the components with `fitContent`.

Right now, alert component is adjusting to the window with a line break and ellipsis:
![image](https://user-images.githubusercontent.com/34685461/191957629-4950b2a0-8e80-4d8d-9b3d-3d31f7224cfb.png)

However, button component is going out from the window:
![image](https://user-images.githubusercontent.com/34685461/191957875-c7d2410a-2e68-4abf-aa80-50b715e09680.png)

Also, it can get out of the container in which it is contained.
What do you think? Should we use ellipsis, line break or any other ideas?

**Comment by Jialecl** [comment]
In my opinion components with a defined height should have an ellipsis (for example the dropdown trigger) while others should jump to the next line.

**Reply by aidamag** [reply]
Well as Ivan said, it is true that ellipsis and line break falls in the scope of design.

**Comment by GomezIvann** [comment, marked as answer]
I think this depends on the component, let me explain myself:

- Box and Alert are the easiest ones in my opinion. They should break on to the next line since they are regular containers used in specific situations. So, just changing fitContent's value to `fit-content` is enough. The ellipsis is a design choice and as far as I know, fit-content should refer to the children's content.
- Regarding the button, I think it is behaving as expected. When the size is `fit-content` it should adapt the width to the label length no matter overflowing. If you read the MDN definition of [fit-content](https://developer.mozilla.org/en-US/docs/Web/CSS/fit-content#:~:text=The%20fit%2Dcontent%20behaves%20as,refer%20to%20the%20content%20size.) it says that when a property has this value, the size goes until max-content, but never more than [max-content](https://developer.mozilla.org/en-US/docs/Web/CSS/max-content). Ellipsis should only apply when the width is smaller than the label, which can happen in every size except fit-content, which must respect the content regardless of whether it overflows.
The rest of the components should behave as the Button (Dropdown, Checkbox, Switch and Tag).

**Reply by aidamag** [reply]
That make sense. I agree on that.
